pkg/controller/backupentry/actuator_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2023 OnMetal authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backupentry
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/gardener/gardener/extensions/pkg/controller/backupentry/genericactuator"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"github.com/go-logr/logr"
+	gomock "github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	. "sigs.k8s.io/controller-runtime/pkg/envtest/komega"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
+
+	commonv1alpha1 "github.com/onmetal/onmetal-api/api/common/v1alpha1"
+	storagev1alpha1 "github.com/onmetal/onmetal-api/api/storage/v1alpha1"
+)
+
+var _ = Describe("BackupEntry Delete", func() {
+	ns := SetupTest()
+
+	var (
+		ctrl               *gomock.Controller
+		a                  genericactuator.BackupEntryDelegate
+		log                logr.Logger
+		mockS3ObjectLister *Mocks3ObjectLister
+	)
+
+	BeforeEach(func(ctx SpecContext) {
+		a = newActuator()
+		Expect(a).NotTo(BeNil())
+		err := a.(inject.Client).InjectClient(k8sClient)
+		Expect(err).NotTo(HaveOccurred())
+
+		ctrl = gomock.NewController(GinkgoT())
+		mockS3ObjectLister = NewMocks3ObjectLister(ctrl)
+
+		objectLister = mockS3ObjectLister
+	})
+
+	It("should delete Backupentry", func(ctx SpecContext) {
+
+		By("creating an Onmetal bucket resource")
+		bucketName := "test-bucket"
+		bucket := &storagev1alpha1.Bucket{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: ns.Name,
+				Name:      bucketName,
+			},
+			Spec: storagev1alpha1.BucketSpec{
+				BucketClassRef: &corev1.LocalObjectReference{
+					Name: "test-bucket-class",
+				},
+				BucketPoolSelector: map[string]string{
+					"key": "value",
+				},
+				BucketPoolRef: &corev1.LocalObjectReference{
+					Name: "my-bucket-pool",
+				},
+				Tolerations: []commonv1alpha1.Toleration{
+					{
+						Key:      "key",
+						Operator: "Equal",
+						Value:    "value",
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, bucket)).Should(Succeed())
+
+		Eventually(Object(bucket)).Should(SatisfyAll(
+			HaveField("Spec.BucketClassRef", &corev1.LocalObjectReference{
+				Name: "test-bucket-class",
+			}),
+		))
```

Comment:
No need to check here for the `classRef` as we are creating it above.

Reply:
removed